components/http/okHttp/src/main/java/com/microsoft/kiota/http/KiotaClientFactory.java

```diff
@@ -92,11 +96,70 @@ private KiotaClientFactory() {}
         };
     }
 
+    /**
+     * Creates the default interceptors for the client.
+     * @return an array of interceptors.
+     */
+    @Nonnull public static Interceptor[] createOveridableDefaultInterceptors( @Nonnull Interceptor[] interceptors) {
```

Comment:
Could still lead to unexpected behaviour since we're always adding the default middleware to the end of the chain. This was a bug in the previous implementation.

Reply:
.NET only uses the custom handlers when provided. It doesn't append default handlers: [ref](https://github.com/microsoft/kiota-dotnet/blob/main/src/http/httpClient/KiotaClientFactory.cs#L54)